src/xo-app/backup/new/index.js

```diff
@@ -183,31 +186,33 @@ export default class New extends Component {
     super(props)
 
     const { state } = this
-    const { job } = props
-
     state.cronPattern = '* * * * *'
+  }
 
-    if (!job) {
+  componentWillMount () {
+    const { job, schedule } = this.props
+    if ((!job && schedule) || (job && !schedule)) {
+      error(_('backupEditNotFoundTitle'), _('backupEditNotFoundMessage'))
+    }
+    if (!job || !schedule) {
```

Comment:
Inverting the order of the `if`s allows for simpler conditions, or you can even nest them:

```js
if (!jobs || !schedule) {

  // if one of them is available, it means we are editing and object is missing
  if (jobs || schedule) {
    error(_('backupEditNotFoundTitle'), _('backupEditNotFoundMessage'))
  }

  return
}
```

Am I right?